src/dataflow/src/render/reduce.rs

```diff
@@ -1092,4 +1154,85 @@ pub mod monoids {
             false
         }
     }
+
+    /// A monoid containing a single-datum row.
```

Comment:
Any reason not to make this "potentially multiple datum" rows? It seems like it would be similar work, but would then simplify the allocation story at least for the fused monotonic stuff.

Reply:
Yeah I think I could wrangle that. The reason I had avoided it is because we already split the incoming row into a vec of rows so that we can in the case of non-monotonic hierarchical aggregates we compute each aggregate over each column without having to iterate through through the row `arity` times. Let me see if I can think of a better way to make that work

Reply:
Further thoughts: this makes the logic for "just copy `lhs` or `rhs`" harder. It also means demultiplexing the mins and maxes because they couldn't be commingled in one `ReductionMonoid` instance.

Reply:
Talked offline and decided this isn't urgent. Filed https://github.com/MaterializeInc/materialize/issues/5007 as a followup